### DISCUSSION TITLE
Retrying a Failed Suspense Data Fetching

### DISCUSSION THREAD

**Discussion by vicnz** [vercel/next.js#44493] [Help]
I wanted to create a Component that uses `next/dynamic` with Suspense wrapped in an `ErrorBoundary` that can be retried whenever the Suspense failed, **(Faced Issue)** but the Suspense Components is not remounting.

```
import {ErrorBoundary} from 'react-error-boundary'
...
const Suspended = dynamic(() => import('path.to.suspending..component'), {suspense: true})

....

return (
 <ErrorBoundary FallbackComponent={Fallback}>
   <Suspense fallback={"loading..."}>
      <Suspended/>
   </Suspense>
</ErrorBoundary>
)
```

Suspended Component

```
const resource = withPromiseWrapper(fetch(''api. endpoint").then(res => res.json()))

...

{
  const data = resource.read()
 return (....)
}
```

###  intended behavior
> when the ErrorBoundary State is *resets* it should trigger a remounting of it's child components and fetching of data by the suspense component *although the (resource) call cannot be called again since it's outside the component, the React Docs specify that it must exists outside the component not inside* which in turn it returns what was initially rendered (fetched) and not in the subsequent retries
 

>And according to the docs `dynamic` function cannot be called inside a component it should be on the top level

## Asking if there is a specific pattern to achieve this

**Comment by musjj** [comment]
Facing the same problem. Did you find a solution?
